source/puppet/5.3/hiera_config_yaml_5.md

```diff
@@ -47,6 +47,25 @@ Related topics: [$confdir][confdir], [`<MODULE>`][module].
 -   `default_hierarchy` - An array of hashes, which sets a default hierarchy to be used only if the normal hierarchy entries do not result in a value. Only allowed in a module's hiera.yaml.
 -   `defaults` - A hash, which can set a default datadir, backend, and options for hierarchy levels.
 
+Example of an environment hiera.yaml file:
+
+```
+---
+version: 5
+
+defaults:
+  datadir: "data"
+
+hierarchy:
+  - name: 'Yaml backend'
+    data_hash: yaml_data
+    paths:
+      - "nodes/%{trusted.certname}.yaml"
+      - 'common.yaml'
```

Comment:
These two entries have different purpose, one is the "node specific" entry, and one is the "common/defaults". They should really be in an entry each.

Reply:
I think that how hierarchy entries are divided depends on the purpose of the entries. I would for instance recommend that an entry that shares common options (an example of this is is an eyaml configuration) will benefit from not repeating such options just to separate paths, even if the paths have a different purpose.